### PR TITLE
Run3-gex108C Backport correct GEM geometry builder in view of demo chambers

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Backport correct GEM geometry builder in view of demo chambers

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Backport the 3 PR's #36715 #36745 #36765 from the master

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by jfernan2** [comment]
+1

**Comment by civanch** [comment]
@cvuosalo , @srimanob, is this PR OK?

**Comment by cvuosalo** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

File: DQM/GEM/interface/GEMOfflineDQMBase.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#ifndef DQM_GEM_GEMOfflineDQMBase_h
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply DQM/GEM/interface/GEMOfflineDQMBase.h:1]: I took from what existed in 12_3_X. I believe it needs to be cleaned up in 12_3_X first. This is simply a backport

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: DQM/GEM/interface/GEMOfflineDQMBase.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (perrotta) [reply DQM/GEM/interface/GEMOfflineDQMBase.h:1]: > I took from what existed in 12_3_X. I believe it needs to be cleaned up in 12_3_X first. This is simply a backport

No, in 12_3 it was **moved** from DQMOffline/Muon, while with this implementation you are duplicating a class, which does not even serve for your backported code. Please remove

**Comment by bsunanda** [comment]
Hi Andrea

I look into CMSSW_12_3_X_2022-01-26-2300 and the classes I copied still exist there. Please let me know when it was moved. Regards

Sunanda
________________________________

File: DQM/GEM/interface/GEMDQMBase.h [hunk 2]
```diff
@@ -28,7 +28,7 @@
 
 class GEMDQMBase : public DQMEDAnalyzer {
 public:
-  // Borrwed from DQMOffline/Muon/interface/GEMOfflineDQMBase.h
+  // Borrwed from DQM/GEM/interface/GEMOfflineDQMBase.h
```
Comment (perrotta) [inline DQM/GEM/interface/GEMDQMBase.h:31]: Please revert (ok, it is a comment line, but let have consistent comments in the release)

File: DQM/GEM/interface/GEMDQMBase.h [hunk 3]
```diff
@@ -568,7 +568,7 @@ class GEMDQMBase : public DQMEDAnalyzer {
   Int_t nNumEtaPartitionGE21_;
 };
 
-// Borrwed from DQMOffline/Muon/interface/GEMOfflineDQMBase.h
+// Borrwed from DQM/GEM/interface/GEMOfflineDQMBase.h
```
Comment (perrotta) [inline DQM/GEM/interface/GEMDQMBase.h:571]: Please revert (ok, it is a comment line, but let have consistent comments in the release)

File: DQM/GEM/interface/GEMDQMBase.h [hunk 4]
```diff
@@ -578,7 +578,7 @@ inline bool GEMDQMBase::checkRefs(const std::vector<T *> &refs) {
   return true;
 }
 
-// The 'get...' functions in the below are borrwed from DQMOffline/Muon/interface/GEMOfflineDQMBase.h
+// The 'get...' functions in the below are borrwed from DQM/GEM/interface/GEMOfflineDQMBase.h
```
Comment (perrotta) [inline DQM/GEM/interface/GEMDQMBase.h:581]: Please revert (ok, it is a comment line, but let have consistent comments in the release)

**Review by perrotta — COMMENTED** [review]
@bsunanda the move was integrated with #36627, entered first in CMSSW_12_3_X_2022-01-14-1100

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
@cvuosalo , @srimanob, is this PR OK after fixes?

**Comment by jfernan2** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

**Comment by cvuosalo** [comment]
+1

File: DQM/GEM/src/GEMDQMBase.cc [hunk 5]
```diff
@@ -26,7 +26,7 @@ int GEMDQMBase::initGeometry(edm::EventSetup const& iSetup) {
   return 0;
 }
 
-// Borrowed from DQMOffline/Muon/src/GEMOfflineDQMBase.cc
+// Borrowed from DQM/GEM/src/GEMOfflineDQMBase.cc
```
Comment (perrotta) [inline DQM/GEM/src/GEMDQMBase.cc:29]:
```suggestion
// Borrowed from DQMOffline/Muon/src/GEMOfflineDQMBase.cc
```

File: DQM/GEM/src/GEMDQMBase.cc [hunk 6]
```diff
@@ -64,7 +66,7 @@ int GEMDQMBase::loadChambers() {
     }
   }
 
-  // Borrwed from DQMOffline/Muon/src/GEMOfflineMonitor.cc
+  // Borrwed from DQM/GEM/src/GEMOfflineMonitor.cc
```
Comment (perrotta) [inline DQM/GEM/src/GEMDQMBase.cc:69]:
```suggestion
  // Borrowed from DQMOffline/Muon/src/GEMOfflineMonitor.cc
```

**Review by perrotta — COMMENTED** [review]
Just in case...

**Comment by cvuosalo** [comment]
+1

**Comment by jfernan2** [comment]
+1
For the records

**Pull request merged**